profiles/mercury.raml

```diff
@@ -159,4 +160,14 @@ validations:
         pattern: ^v[0-9]+$
         minCount: 1
 
+  base-uri-matches-pattern:
+    message: The baseUri must match the pattern - https://{shortCode}.api.commercecloud.salesforce.com/<api-family>/<api-name>/{version}
+    targetClass: apiContract.WebAPI
+    functionConstraint:
+      code: |
+        function(resource) {
```

Comment:
Why code here instead of a pattern regex enforcement?

Reply:
To validate that a baseUri is part of the spec. `apiContract.urlTempalte` is nested inside `apiContract.Server`. When baseUri is not present in the raml, it doesn't even execute propertyConstraints.